ddtrace/_trace/trace_handlers.py

```diff
@@ -121,11 +125,22 @@ def _start_span(ctx: core.ExecutionContext, call_trace: bool = True, **kwargs) -
     distributed_context = ctx.get_item("distributed_context")
     if distributed_context and not call_trace:
         span_kwargs["child_of"] = distributed_context
+
+    # dispatch event for checking headers and possibly making an inferred proxy span
+    core.dispatch("inferred_proxy.start", (ctx, tracer, span_kwargs, call_trace, distributed_headers_config))
```

Comment:
If we moved the check for `if not config._inferred_proxy_services_enabled:` all the way up to here, we could avoid doing this unnecesarily for every non api-gateway span: `span_kwargs = ctx.get_item("span_kwargs", span_kwargs)`

Reply:
make sense, I moved that check to here